engine/textures/IgeTileMap2dSmartTexture.ts

```diff
@@ -81,9 +79,9 @@ export const IgeTileMap2dSmartTexture: IgeSmartTexture = {
 
 		if (ent._highlightOccupied) {
 			ctx.fillStyle = "#ff0000";
-			for (y in ent.map._mapData) {
+			for (y of ent.map._mapData.keys()) {
 				if (ent.map._mapData[y]) {
-					for (x in ent.map._mapData[y]) {
+					for (x of ent.map._mapData[y].keys()) {
```

Comment:
Does this actually work? I would have thought you'd need to use Object.keys() instead? Where does the .keys() method come from?

Reply:
Yep! It was new-ish to me but apparently it's been around a while: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/keys

Reply:
Yeah I wasn't aware of that function but I guess it makes sense when dealing with sparse arrays. Nice to learn new things! :)